api/types.go

```diff
@@ -217,11 +217,45 @@ type ToolFunction struct {
 		Properties map[string]struct {
 			Type        PropertyType `json:"type"`
 			Description string       `json:"description"`
-			Enum        []string     `json:"enum,omitempty"`
+			Enum        []any        `json:"enum,omitempty"`
 		} `json:"properties"`
 	} `json:"parameters"`
 }
 
+// https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-validation-01#section-6.1.2
+func (t *ToolFunction) UnmarshalJSON(b []byte) error {
+	type Alias ToolFunction
+	var a Alias
+	if err := json.Unmarshal(b, &a); err != nil {
+		return err
+	}
+
+	// Validate enum arrays
+	for _, prop := range a.Parameters.Properties {
+		if len(prop.Enum) == 0 {
+			continue
+		}
+
+		// Check for uniqueness and consistent types
```

Comment:
Is this necessary? Tools should be opaque to Ollama. The only reason any fields are defined is control templating and/or parsing

Reply:
I suppose it doesn't have to be. To me it made more sense to have the JSON validation here vs. the SDKs/client. I don't think we have to be so strict for this as the model would consume whatever we pass it (even if incorrect from a JSON standpoint)